pkg/service/service.go

```diff
@@ -14,34 +14,44 @@ import (
 	"github.com/livekit/egress/pkg/config"
 	"github.com/livekit/egress/pkg/stats"
 	"github.com/livekit/egress/version"
+	"github.com/livekit/livekit-server/pkg/service/rpc"
 	"github.com/livekit/protocol/egress"
 	"github.com/livekit/protocol/livekit"
 	"github.com/livekit/protocol/logger"
 	"github.com/livekit/protocol/tracer"
+	"github.com/livekit/psrpc"
 )
 
 const shutdownTimer = time.Second * 30
 
 type Service struct {
-	conf       *config.ServiceConfig
-	rpcServer  egress.RPCServer
-	promServer *http.Server
-	monitor    *stats.Monitor
-	manager    *ProcessManager
+	conf        *config.ServiceConfig
+	rpcServerV0 egress.RPCServer
+	psrpcServer rpc.EgressInternalServer
+	promServer  *http.Server
+	monitor     *stats.Monitor
+	manager     *ProcessManager
 
 	shutdown chan struct{}
 }
 
-func NewService(conf *config.ServiceConfig, rpcServer egress.RPCServer) *Service {
+func NewService(conf *config.ServiceConfig, bus psrpc.MessageBus, rpcServerV0 egress.RPCServer) (*Service, error) {
 	monitor := stats.NewMonitor()
 
 	s := &Service{
-		conf:      conf,
-		rpcServer: rpcServer,
-		monitor:   monitor,
-		manager:   NewProcessManager(conf, monitor),
-		shutdown:  make(chan struct{}),
+		conf:        conf,
+		rpcServerV0: rpcServerV0,
+		monitor:     monitor,
+		manager:     NewProcessManager(conf, monitor),
+		shutdown:    make(chan struct{}),
 	}
+
+	psrpcServer, err := rpc.NewEgressInternalServer(conf.NodeID, s, bus)
+	if err != nil {
+		return nil, err
+	}
+	s.psrpcServer = psrpcServer
```

Comment:
how is this used? I don't see references to psrpcServer anywhere.. does it just get started automatically?

Reply:
added a shutdown method to drain and close

Reply:
so it does start automatically when the server is created?

Reply:
yeah